File: Corte1/Tareas/RetoTienda.cpp
#include <iostream>
using namespace std;

int main() {

    int productos, cantidad;
    double precio, subtotal;
    double total = 0;
    double descuento;
    double total_pagar;
    string nombre;

    cout << "Cuantos productos va a comprar? ";
    cin >> productos;

    for (int i = 1; i <= productos; i++) {

        cout << "\nProducto " << i << endl;

        cout << "Nombre del producto: ";
        cin >> nombre;

        cout << "Precio unitario: ";
        cin >> precio;

        cout << "Cantidad comprada: ";
        cin >> cantidad;

        subtotal = precio * cantidad;

        total = total + subtotal;

        cout << "Subtotal de " << nombre << ": " << subtotal << endl;
    }

    if (total > 300000) {

        descuento = total * 0.10;

    } else if (total >= 150000) {

        descuento = total * 0.05;

    } else {

        descuento = 0;
    }

    total_pagar = total - descuento;

    cout << "\n-----------------------------" << endl;
    cout << "Total antes del descuento: " << total << endl;
    cout << "Descuento aplicado: " << descuento << endl;
    cout << "Total a pagar: " << total_pagar << endl;

    return 0;
}

File: Corte1/Tareas/RetoTienda.py
productos = int(input("¿Cuántos productos va a comprar? "))

total = 0

for i in range(productos):

    nombre = input("Nombre del producto: ")
    precio = float(input("Precio unitario: "))
    cantidad = int(input("Cantidad comprada: "))

    subtotal = precio * cantidad

    total = total + subtotal

    print("Subtotal de", nombre, ":", subtotal)


if total > 300000:
    descuento = total * 0.10

elif total >= 150000:
    descuento = total * 0.05

else:
    descuento = 0


total_pagar = total - descuento

print("-------------------------")
print("Total antes del descuento:", total)
print("Descuento aplicado:", descuento)
print("Total a pagar:", total_pagar)